src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -417,13 +417,43 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             vkey != VK_SNAPSHOT &&
             keyDown)
         {
-            const auto isInMarkMode = _terminal->SelectionMode() == ::Microsoft::Terminal::Core::Terminal::SelectionInteractionMode::Mark;
-            if (isInMarkMode && modifiers.IsCtrlPressed() && vkey == 'A')
+            const auto isInMarkMode = _terminal->SelectionMode() == ::Terminal::SelectionInteractionMode::Mark;
+            if (isInMarkMode)
             {
-                auto lock = _terminal->LockForWriting();
-                _terminal->SelectAll();
-                _updateSelectionUI();
-                return true;
+                if (modifiers.IsCtrlPressed() && vkey == 'A')

```

Comment:
FWIW: I feel like this is below a "dotted line" boundary where ControlInteractivity and ControlCore meet. This is all interactive stuff, and all specifies the interaction model for the control. @zadjii-msft may be more (or less?) opinionated about this, but like... it feels like we're blending the layers in a way that is not mutually beneficial to all the components.

We should discuss!

Reply:
oof, just ran into this: https://github.com/microsoft/terminal/blob/574a72b505a682a09ea9d533295d3d576556e6c0/src/cascadia/TerminalControl/ControlCore.cpp#L1606-L1616

I think this may just be another example of the dotted lines not working right. :/

Reply:
Maybe this is a case of bad naming. In my head, both WPF and UWP controls would use Interactivity as the owner for the core. I'm not sure there's a world where they'd be separate. 

Curiously, `TrySendKeyEvent` doesn't exist on `Interactivity`, it only exists on `Core`. That's kinda weird. Probably an artifact of how the control/core refactor happened - moving this to the right place wasn't _needed_ so it was left in a weird place. _In an ideal world_:
* yea `TrySendKeyEvent` should be exposed off interactivity, with this selection code in it
* Core should expose to Interactivity stuff like `SelectAll`, `SelectHyperlink`, `TryOpenSelectedHyperlink` (ew), etc.


I think. But I wouldn't call this blocking.